File: Remove and Reverse - GFG/remove-and-reverse.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends
#User function Template for python3
class Solution:
    def removeReverse(self, s): 
        d={}
        n=len(s)
        s=list(s)
        for i in s:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        i=0
        j=n-1
        di=0
        flag=0
        while(i<j):
            if(di==0):
                k=s[i]
                if(d[k]>1):
                    d[k]-=1
                    s[i]='@'
                    
                    di=1
                i+=1
                
            else:
                k=s[j]
                if(d[k]>1):
                    d[k]-=1
                    s[j]='@'
                    
                    di=0
                j-=1
                
            
        ans=""
        for i in range(n):
            if(s[i]!='@'):
                ans+=s[i]
        
        if(di==0):
            return ans
        return ans[::-1]

        #code here
        

#{ 
 # Driver Code Starts.
if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        S = input()
        ob = Solution()
        ans = ob.removeReverse(S)
        print(ans)


# } Driver Code Ends

File: Remove and Reverse - GFG/remove-and-reverse.cpp
//{ Driver Code Starts
//Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++
class Solution {
  public:
    string removeReverse(string S) {
        // code hre
        map<char,int>m;
        for(auto i : S){
            m[i]++;
        }
        int n=S.size();
        int i=0,j=n-1,d=0,k=0;
       while(i<j){
           if(d==0){
               k=i;
               i++;
               
           }
           else{
               k=j;
               j--;
               
           }
       
           if(m[S[k]]>1){
            m[S[k]]--;
            S[k]='@';
            if(d==0) d=1;
            else d=0;
           } 
           
        }
        string ans="";
        for(auto i : S){
            if(i!='@') ans+=i;
        }
        if(d==0) return ans;
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        string S;
        cin>>S;
        Solution obj;
        auto ans = obj.removeReverse(S);
        cout<<ans<<endl;
    }
}
// } Driver Code Ends